internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -218,6 +225,73 @@ func (a *Adapter) getComponentSourceFromPipelineRun(pipelineRun *tektonv1.Pipeli
 	return &componentSource, nil
 }
 
+// notifySnapshotsInGroupAboutFailedBuild tries to find the latest group Snapshot and notify it about the failed build.
+// If the group Snapshot can't be found, find the component Snapshots that would belong to the same group and notify them instead.
+func (a *Adapter) notifySnapshotsInGroupAboutFailedBuild(pipelineRun *tektonv1.PipelineRun) error {
+	prGroupHash := pipelineRun.Labels[gitops.PRGroupHashLabel]
+	buildPLRFailureMsg := fmt.Sprintf("build PLR %s failed for component %s so it can't be added to the group Snapshot for PR group %s", pipelineRun.Name, a.component.Name, prGroupHash)
+	notified, err := a.notifyComponentSnapshotsInLatestGroupSnapshot(pipelineRun, prGroupHash, buildPLRFailureMsg)
```

Comment:
I am thinking if we need to notify component snapshots in latest group snapshot here, especially it `return nil` when `notified`. I feel the component snapshots in latest group snapshot is not the latest snapshot for each component.
Once a group snapshot is created, that means one cycle of testing for that pr group has been finished. These component snapshots don't need to know the status of later failing build PLR.
I feel `allComponentSnapshotsInGroup, err := a.loader.GetMatchingSnapshotsForPRGroupHash(a.context, a.client, pipelineRun.Namespace, prGroupHash, gitops.SnapshotComponentType)` is enough. Maybe I am wrong.

Reply:
But others look good to me.

Reply:
It reminds me, we need to also to fix the same issue mentioned in [KFLUXBUGS-1761](https://issues.redhat.com/browse/KFLUXBUGS-1761) for group snapshot.

Reply:
That's a good point, I updated the code to only notify the latest component Snapshots belonging to that group (latest per-component), as well as to annotate in-flight build pipelineRuns as before.

Reply:
Thank you!